schemas/input/csl-citation.json

```diff
@@ -15,8 +15,13 @@
       "pattern": "1.1"
     },
     "citationID": {
+      "title": "Citation ID",
+      "description": "A stable and unique ID (for example, a UUID) assigned to the citation, for internal use by the processor.",
       "type": ["string", "number"]
     },
+    "noteIndex": {
+      "type": "number"
```

Comment:
Shouldn't this be an integer @fbennett?

Reply:
In theory, at least, the ``noteIndex`` isn't used to determine sequence, it's just the marker associated with the note in which the reference is placed. Again in theory, the marker might be ``*`` (for the author bio in the first note), or numbers might be reset at the start of a new chapter, and ``noteIndex`` should carry those literal values for back-referencing purposes. Sequence (for the purpose of evaluating ``ibid` etc.) can be determined from the order in which citations are submitted to the processor.

Unfortunately (from memory--this might not be totally accurate), this doesn't work out so nicely in practice. IIRC, Zotero's integration plugins supply the *sequence* number of the note in ``noteIndex``, since for technical reasons extracting the note marker from some word processors is difficult. With an initial author bio in a ``*`` note, the value of ``first-reference-note-number`` (derived from ``noteIndex`` is thrown off; and therefore the recommended practice (worked out by @adam3smith) is to set the initial author bio as a footer that spoofs the appearance of a footnote.

For a consolidated doc that resets note numbering at the start of each chapter, the problems are more severe. There is no indication in the citation input of the chapter in which the note occurs, so the behavior of ``first-reference-note-number`` can best be described as "undefined" (I'm not actually sure what happens in such a document, but it wouldn't be pretty).

If the status quo is acceptable, though, constraining the value of ``noteIndex`` to positive integer values would be fine.

Reply:
Not sure whether we should the change, but we should at least add an annotation. 

This is copy-and-pasted from your docs:

> Indicates the footnote number in which the citation is located within the document.

This is why I was assuming a positive integer, but I see that's actually not the case. 

So maybe we revise it to this, and allow a string too?

> Indicates the footnote number, character, or symbol in which the citation is located within the document.

Reply:
Also, are you OK with me removing `properties` and putting `noteIndex` alongside `citationID` instead, per #347?